pkg/controller/update_cluster.go

```diff
@@ -532,12 +534,25 @@ func (c *M3DBController) expandPlacementForSet(
 		}
 	}
 	if len(podsToAdd) == 0 {
-		return errors.New("could not find pod absent from placement")
+		c.logger.Info("pods missing from placement are not indexed yet.")
```

Comment:
again another indexer delay that might be a real issue in prod.

Reply:
One nice benefit of returning an error here is that it'll immediately trigger a requeue of the cluster event loop. If we return nil, it's likely that the event that causes the indexer to pick up the pod will trigger a reconcile, but I'm not 100% sure.

The sooner we re-reconcile the sooner the cluster converges on the desired state. Otherwise you have to wait for the requeue timeout (5min), or an unrelated event.

Reply:
ah good point. I'll change it back to an error.

I guess the tests will just need to allow these of transient errors.